### PR TITLE
Cleaned Up CheckpointTest

### PR DESCRIPTION
**Pull request by Maithem**
## Overview
Made all the tests thread safe.

Why should this be merged: Fixes some tests that have been failing intermittently 

Related issue(s) (if applicable): #1527

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java [hunk 1]
```diff
@@ -130,10 +116,10 @@ void mapCkpointAndTrim() throws Exception {
      * @param expectedFullsize
      */
     void validateMapRebuild(int mapSize, boolean expectedFullsize) {
-        setRuntime();
+        CorfuRuntime rt = getARuntime();
```
[the remark this answers is not in the document]
Comment (Maithem) [reply test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java]: I see, I think the test intended to simulate new clients seeing the objects.

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java [hunk 2]
```diff
@@ -186,15 +174,20 @@ void populateMaps(int mapSize) {
...
+
+        Map<String, Long> mapA = openMap(rt, streamNameA);
+        Map<String, Long> mapB = openMap(rt, streamNameB);
+
         // thread 1: pupolates the maps with mapSize items
         scheduleConcurrently(1, ignored_task_num -> {
-                    populateMaps(mapSize);
+                    populateMaps(mapSize, mapA, mapB);
                 });
```
[the remark this answers is not in the document]
Comment (Maithem) [reply test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java]: done.

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Maithem) [reply test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java]: How many is too many? I see that the number of iterations is low. How many client creations are yous seeing ?

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (xnull) [reply test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointTest.java]: ignoredTaskNum

**Pull request merged**